doc/radosgw/rgw-cache.rst

```diff
@@ -0,0 +1,51 @@
+========================== 
+RGW Data caching and CDN
+==========================
+
+.. versionadded:: Octopus
+
+.. contents::
+
+This new feature is intend to add to RGW, new ability to cache objects securely and offload the workload from the cluster, using Nginx.
+After an object is accessed the first time it will be stored on top of Nginx dir.
+Every request the Nginx will not fetch data from the RGW or the cluster, It will only check for permissions for that user from the RGW.
+This feature is based on some Nginx modules, ngx_http_auth_request_module, https://github.com/kaltura/nginx-aws-auth-module, Openresty for lua capablities.     
+Currently this feature only works for GET requests and it will cache only AWSv4 requests (only s3 requests).
+The feature introduces 2 new APIs: Auth and Cache.
+
+New APIs
+-------------------------
```

Comment:
from a user perspective, i don't think it's helpful to document the APIs themselves. users just want to know what it does and how to set it up

i also don't think ceph docs are the right place to put these scripts and config files. since this is about layering another technology on top of radosgw, i'd rather put that stuff in a separate git repo with its own documentation

Reply:
I still need to refine the docs, but yes we will need to find a way storing config files (or maybe building ansible operator for openshift deployment)

Reply:
Updated the docs, I want your feedback how to improve them to make the feature usable.
@mykaul @mattbenjamin @cbodley